scripts/configure-dh.sh

```diff
@@ -48,465 +53,478 @@ KUBERNETES_SA=${KUBERNETES_SA:-'rhdh-kubernetes-plugin'}
+        until [ ! -z "${GITHUB__ORG__NAME}" ]; do
+            read -p "Enter your GitHub Org Name: " GITHUB__ORG__NAME
+            if [ -z "${GITHUB__ORG__NAME}" ]; then
+                echo "No GitHub Org Name entered, try again."
             fi
         done
     fi
-fi
 
-if [[ ! $RHDH_SIGNIN_PROVIDER =~ ^(github|gitlab)$ ]]; then 
-    RHDH_SIGNIN_PROVIDER="${signin_provider}"
-fi
+    # Reads GitLab secrets if enabling GitHub integration
+    if [[ $RHDH_GITLAB_INTEGRATION == "true" ]]; then
+        # Reads GitLab App Client ID
+        until [ ! -z "${GITLAB__APP__CLIENT__ID}" ]; do
+            read -p "Enter your GitLab App Client ID: " GITLAB__APP__CLIENT__ID
+            if [ -z "${GITLAB__APP__CLIENT__ID}" ]; then
+                echo "No GitLab App Client ID entered, try again."
+            fi
+        done
+
+        # Reads GitLab App Client Secret
+        until [ ! -z "${GITLAB__APP__CLIENT__SECRET}" ]; do
+            read -p "Enter your GitLab App Client Secret: " GITLAB__APP__CLIENT__SECRET
+            if [ -z "${GITLAB__APP__CLIENT__SECRET}" ]; then
+                echo "No GitLab App Client Secret entered, try again."
+            fi
+        done
+
+        # Reads GitLab PAT
+        until [ ! -z "${GITLAB__TOKEN}" ]; do
+            read -p "Enter your GitLab Token: " GITLAB__TOKEN
+            if [ -z "${GITLAB__TOKEN}" ]; then
+                echo "No GitLab Token entered, try again."
+            fi
+        done
 
-# Reads Quay API Token
-# Optional: If an API Token is not entered, there will be none provided to the developer hub app config
-if [[ ! $BYPASS_OPTIONAL_INPUT =~ ",QUAY__API_TOKEN" ]] && [ -z "${QUAY__API_TOKEN}" ]; then
-    read -p "Enter your Quay API Token (Optional): " QUAY__API_TOKEN
-fi
+        # Reads GitLab Group Name
+        until [ ! -z "${GITLAB__GROUP__NAME}" ]; do
+            read -p "Enter your GitLab Group Name: " GITLAB__GROUP__NAME
+            if [ -z "${GITLAB__GROUP__NAME}" ]; then
+                echo "No GitLab Group Name entered, try again."
+            fi
+        done
 
-echo "OK"
+        # Reads GitLab Org Enabled
+        if [ -z "${GITLAB__ORG__ENABLED}" ] && [[ $GITLAB__HOST == "gitlab.com" ]]
+        then
+            GITLAB__ORG__ENABLED='true' # required for gitlab.com, see https://backstage.io/docs/integrations/gitlab/org#users
+        elif [ -z "${GITLAB__ORG__ENABLED}" ]
+        then
+            prompt=''
+            until [[ "${GITLAB__ORG__ENABLED}" == "true" ]] || [[ "${GITLAB__ORG__ENABLED}" == "false" ]]; do
+                read -p "Is GitLab Organizations enabled? (y/n): " prompt
+
+                case "$prompt" in
+                    y)
+                        GITLAB__ORG__ENABLED='true'
+                        ;;
+                    n)
+                        GITLAB__ORG__ENABLED='false'
+                        ;;
+                    *)
+                        echo 'Please enter "y" or "n", try again.'
+                        ;;
```

Comment:
```suggestion
                    y)
                        GITLAB__ORG__ENABLED='true';;
                    n)
                        GITLAB__ORG__ENABLED='false';;
                    *)
                        echo 'Please enter "y" or "n", try again.';;
```

if my bash serves me right I think we can move the `;;` up inline with the logic to cut out some excess lines from the long script, wdyt?

Reply:
If that works yeah I don't having an issue with that, especially with them being one liners.